changelog/2024-03-04T17_11_12+01_00_fix_numeric_issues

```diff
@@ -0,0 +1 @@
+FIXES: various issues with primitives on various number types, see PRxxxx for details
```

Comment:
```suggestion
FIXES: various issues with primitives on various number types
```

[blame-pr.py](https://github.com/clash-lang/clash-compiler/blob/master/changelog/blame-pr.py) will handle it once we make the release log

Reply:
Done.
Is the rest ok?
It's too vague for my liking. But I couldn't find every thing in one sentence.

Reply:
Maybe:

```
FIXED: various issues with black boxes and evaluator rules implemented for GHC's number-related primitives
```

?

Reply:
Thnx for the suggestion.
I've taken out the GHC part, because it also fixes things for `Signed` and `Unsigned`.